pkg/cmd/roachtest/tests/failover.go

```diff
@@ -640,6 +715,26 @@ func makeFailer(
 	}
 }
 
+// Disconnect takes a set of nodes and each nodes internal ips. It disconnects
+// each node from all the others in the list.
+func Disconnect(t test.Test, c cluster.Cluster, ctx context.Context, nodes []int, reconnect bool) {
+	ips, err := c.InternalIP(ctx, t.L(), nodes)
+	require.NoError(t, err)
+
+	// disconnect each node from every other passed in node.
+	for n := 0; n < len(nodes); n++ {
+		for ip := 0; ip < len(ips); ip++ {
+			if n != ip {
+				if reconnect {
+					c.Run(ctx, c.Node(nodes[n]), `sudo iptables -F`)
```

Comment:
We'll need to run this cleanup on all nodes whenever the test fails too, since the cluster can be reused for other roachtests which would otherwise get weird failures because networking is broken. A `defer` should do it.